### PR TITLE
chore(rust): migrate workspace to edition 2024

### PR DESCRIPTION
**Pull request by gh-xj**
## Summary

- What changed?
  - Migrated the workspace and rustfmt config from Rust 2021 to Rust 2024.
  - Cleared Rust 2024 compatibility blockers across app/kernel/protocol/spec.
  - Removed process-global env mutation from memory config setup and switched to injected `MemoryRuntimeConfig`.
  - Renamed the `gen` identifier blocked by the 2024 keyword set.
  - Made drop order explicit where 2024 tail-expression behavior changes mattered.
  - Accepted the repo-wide rustfmt 2024 formatting churn and cleaned strict clippy findings exposed during the migration.
  - Rebased the migration onto the updated `alpha-test` branch and preserved its memory-window hardening changes.

- Why this change is needed?
  - The repo was still pinned to Rust 2021 even though toolchain/CI were already on `stable`.
  - Rust 2024 migration needed dedicated compatibility work before the edition flip.
  - Doing the migration now removes known edition blockers and keeps the workspace aligned with current toolchain behavior.

## Scope

- [ ] Small and focused
- [ ] Includes docs updates (if needed)
- [x] No unrelated refactors

## Risk Track

- [ ] Track A (routine/low-risk)
- [x] Track B (higher-risk/policy-impacting)

If Track B, include design/risk notes:
- This PR changes the workspace edition and pulls in repo-wide rustfmt 2024 formatting churn.
- Memory runtime config is now process-global via `OnceLock` instead of env mutation at bootstrap. That is the intended migration path here, but it means the first init in a process wins.
- During verification, a transient `loongclaw-daemon` `SIGABRT` was observed only when multiple `cargo test` commands were run concurrently against the same target/artifact dir. The exact CI-parity commands below passed when run sequentially.

## Validation

- [x] `cargo fmt --all -- --check`
- [x] `cargo clippy --workspace --all-targets --all-features -- -D warnings`
- [x] `cargo test --workspace --all-features`
- [x] Additional scenario/benchmark checks (if applicable)
- [x] If this changes config/env fallback, limits, or defaults: include before/after behavior and regression coverage for explicit path, fallback path, and boundary values
- [x] If tests mutate process-global env: document how state is restored or serialized

Additional validation details:
- Ran `cargo clippy --workspace --all-targets --all-features -- -W rust-2024-compatibility`
- Ran `cargo test --workspace`
- Explicit path behavior: injected sqlite path still routes memory IO to the requested DB path
- Fallback path behavior: sqlite fallback path still resolves under the default loongclaw home
- Boundary values: sliding window parsing/validation covers absent, invalid, zero, explicit positive, and upper-bound clamp behavior
- Process-global env mutation in tests was removed rather than serialized; the old helper was deleted and home-resolution coverage is now deterministic/pure-function based

## Linked Issues

Closes #

### CODE REVIEW

File: crates/app/src/memory/sqlite.rs [hunk 1]
```diff
@@ -80,13 +80,15 @@ pub(super) fn load_window(
...
-        .unwrap_or_else(default_window_size_u64)
-        .clamp(1, 128) as usize;
-    let default_window = default_window_size().max(1);
-    let window_limit = requested_limit.min(default_window);
+        .map(|limit| limit as usize)
+        .unwrap_or_else(|| default_window_size(config))
+        .clamp(
+            crate::config::MIN_MEMORY_SLIDING_WINDOW,
+            crate::config::MAX_MEMORY_SLIDING_WINDOW,
+        );
```
[the remark this answers is not in the document]
Comment (gh-xj) [reply crates/app/src/memory/sqlite.rs]: Updated in ca4861e and now on PR head f95eb7d: the window limit is clamped in u64 before conversion, and the hard cap literals in crates/app/src/memory/sqlite.rs are explicitly typed as u64. That keeps large inputs from truncating before bounds are enforced.
